Adds "Elsa.OpenAI" capabilities

Zettersten opened the pull request.

Zettersten commented:
@dotnet-policy-service agree

sfmskywalker commented on src/Elsa.OpenAI/README.md:
```diff
@@ -0,0 +1,310 @@
+                    MaxTokens = new(280),
+                    Temperature = new(0.8f),
+                    Result = new(socialPost)
+                },
```
@copilot open a new pull request to apply changes based on [this feedback](https://github.com/elsa-workflows/elsa-extensions/pull/98#discussion_r2603737114)

Zettersten commented:
Hey @sfmskywalker 👋 — this PR has been updated and is ready for review.

**What changed since the original submission:**
- **Rebased onto current main** — clean single commit, no merge conflicts
- **Bumped OpenAI SDK** from 2.7.0 → 2.10.0
- **Moved to module structure** — now at src/modules/llm/Elsa.OpenAI and 	est/modules/llm/Elsa.OpenAI.Tests following upstream conventions
- **Applied .NET best practices** — sealed types, ConcurrentDictionary for thread-safe client caching, ArgumentException.ThrowIfNullOrWhiteSpace, proper ChatMessage list usage, comprehensive XML docs
- **67 passing tests** with parameterized validation coverage
- **Fixed README examples** — all CompleteChat instances now include the required ApiKey property

Happy to make any adjustments based on your feedback!

Zettersten committed:
feat: Add Elsa.OpenAI integration module under src/modules/llm

- OpenAI SDK 2.10.0 with thread-safe ConcurrentDictionary-based client factory

- Activities: CompleteChat with proper ChatMessage list and ChatCompletionOptions

- Module structure follows upstream conventions (src/modules/llm, test/modules/llm)

- Comprehensive test suite (67 tests) with parameterized validation coverage

- README with complete working examples including ApiKey configuration

Co-Authored-By: Oz <oz-agent@warp.dev>
